http/middleware.go

```diff
@@ -36,6 +37,29 @@ func NewAuthMiddleware(provider eduboard.UserAuthenticationProvider) func(handle
 	}
 }
 
+func BasicAuth(username, password string) func(handler http.Handler) http.Handler {
```

Comment:
Let's discuss in the group but I don't understand why we need a different authentication for admins here.

Reply:
Clarified. Admins are treated independently from teachers and students, thus get a different auth method as well. 👍